examples/solvers/chambolle_pock_deconvolve.py

```diff
@@ -0,0 +1,141 @@
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Deconvolution using Chambolle-Pock solver."""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+
+standard_library.install_aliases()
+from builtins import super
+
+# External
+import numpy as np
+import scipy
+import scipy.ndimage
+import matplotlib.pyplot as plt
+
+# Internal
+import odl
+
+
+# TODO: Use BroadCastOperator instead of ProductSpaceOperator
+
+class Convolution(odl.Operator):
+    def __init__(self, space, kernel, adjkernel):
+        self.kernel = kernel
+        self.adjkernel = adjkernel
+        super().__init__(space, space, linear=True)
+
+    def _call(self, rhs, out):
+        scipy.ndimage.convolve(rhs, self.kernel, output=out.asarray(),
+                               mode='wrap')
+        return out
+
+    @property
+    def adjoint(self):
+        return Convolution(self.domain, self.adjkernel, self.kernel)
+
+
+def kernel(x):
+    mean = [0.1, 0.5]
+    std = [0.1, 0.1]
+    k = np.exp(-(((x[0] - mean[0]) / std[0]) ** 2 +
+                 ((x[1] - mean[1]) / std[1]) ** 2))
+    k = np.around(k, 8)
+    return k
+
+
+def adjkernel(x):
+    return kernel((-x[0], -x[1]))
+
+
+# Continuous definition of problem
+cont_space = odl.FunctionSpace(odl.Rectangle([-1, -1], [1, 1]))
+kernel_space = odl.FunctionSpace(cont_space.domain - cont_space.domain)
+
+# Discretization parameters
+n = 50
+npoints = np.array([n + 1, n + 1])
+npoints_kernel = np.array([2 * n + 1, 2 * n + 1])
+
+# Discretized spaces
+discr_space = odl.uniform_discr_fromspace(cont_space, npoints)
+discr_kernel_space = odl.uniform_discr_fromspace(kernel_space, npoints_kernel)
+
+# Discretize the functions
+disc_kernel = discr_kernel_space.element(kernel)
+disc_adjkernel = discr_kernel_space.element(adjkernel)
+
+# Initialize convolution operator
+conv = Convolution(discr_space, disc_kernel, disc_adjkernel)
+
+# Optional: Run diagnostics to assure the adjoint is properly implemented
+# odl.diagnostics.OperatorTest(conv_op).run_tests()
+
+# Initialize gradient operator
+grad = odl.Gradient(discr_space, method='forward')
+
+# Matrix of operators
+op = odl.ProductSpaceOperator([[conv], [grad]])
+
+# Load phantom
+discr_phantom = odl.util.phantom.shepp_logan(discr_space, modified=True)
+
+# Create data vector of convolved phantom
+data = conv(discr_phantom)
+
+# Starting point
+x = op.domain.one()
+
+# Operator norm, add 10 percent to ensure ||K||_2^2 * sigma * tau < 1
+op_norm = 1.1 * odl.operator.oputils.power_method_opnorm(op, 100)
```

Comment:
Perhaps we should expose this method to the main namespace? It's quite useful and the name is long enough to avoid confusion.